build/openresty/wasmx/rules.bzl

```diff
@@ -0,0 +1,88 @@
+load("//build/openresty/wasmx:wasmx_repositories.bzl", "wasm_runtimes")
+
+wasmx_configure_options = select({
+    "@platforms//os:linux": [
+        # neded for wasmx module
+        # although this is centos7 specific, the flag will work on any GNU linker
+        # we place it here to skip macos, which uses darwin ld
+        # https://github.com/Kong/ngx_wasm_module/commit/e70a19f53e1dda99d016c5cfa393652720959afd
+        "--with-ld-opt=\"-Wl,--allow-multiple-definition\"",
+    ],
+    "//conditions:default": [],
+}) + select({
+    "@kong//:wasmx_flag": [
+        "--with-cc-opt=\"-DNGX_WASM_HOST_PROPERTY_NAMESPACE=kong\"",
+        "--with-cc-opt=\"-I$$INSTALLDIR/../wasm_runtime/include\"",
+        "--with-ld-opt=\"-L$$INSTALLDIR/../wasm_runtime/lib\"",
+    ],
+    "//conditions:default": [],
+}) + select({
+    "@kong//:wasmx_static_mod": [
+        "--add-module=$$EXT_BUILD_ROOT$$/external/ngx_wasm_module",
+    ],
+    "@kong//:wasmx_dynamic_mod": [
+        "--with-compat",
+        "--add-dynamic-module=$$EXT_BUILD_ROOT$$/external/ngx_wasm_module",
+    ],
+    "//conditions:default": [],
+})
+
+wasmx_env = select({
+    "@kong//build/openresty/wasmx:use_v8": {
+        "NGX_WASM_RUNTIME": "v8",
+        # https://github.com/Kong/ngx_wasm_module/blob/0f07c712c48d410190ec5e0cc0b34fdfd190387d/t/10-build/003-dynamic_module.t#L43
+        "NGX_WASM_RUNTIME_LD_OPT": "$$INSTALLDIR/../wasm_runtime/lib/libwee8.a -lv8bridge -lstdc++ -lm -ldl -lpthread",
+        # see the above comments and source for this dummy ar script
+        "AR": "$(execpath @kong//build/openresty:wasmx/wasmx_v8_ar)",
+    },
+    "@kong//build/openresty/wasmx:use_wasmer": {
+        "NGX_WASM_RUNTIME": "wasmer",
+        # https://github.com/Kong/ngx_wasm_module/blob/0f07c712c48d410190ec5e0cc0b34fdfd190387d/t/10-build/003-dynamic_module.t#L30
+        "NGX_WASM_RUNTIME_LD_OPT": "$$INSTALLDIR/../wasm_runtime/lib/libwasmer.a -lm -ldl -lpthread",
+    },
+    "@kong//build/openresty/wasmx:use_wasmtime": {
+        "NGX_WASM_RUNTIME": "wasmtime",
+        # https://github.com/Kong/ngx_wasm_module/blob/0f07c712c48d410190ec5e0cc0b34fdfd190387d/t/10-build/003-dynamic_module.t#L30
+        "NGX_WASM_RUNTIME_LD_OPT": "$$INSTALLDIR/../wasm_runtime/lib/libwasmtime.a -lm -ldl -lpthread",
+    },
+    "//conditions:default": {},
+}) | select({
+    "@kong//:wasmx_flag": {
+        "NGX_WASM_RUNTIME_LIB": "$$INSTALLDIR/../wasm_runtime/lib",
+        "NGX_WASM_RUNTIME_INC": "$$INSTALLDIR/../wasm_runtime/include",
```

Comment:
This one is also redundant with the above `--with-cc-opt` option which already includes this path.